packages/webvh/src/dids/WebvhDidRegistrar.ts

```
@@ -0,0 +1,239 @@
+        },
+      }
+    } catch (error) {
+      agentContext.config.logger.error(`Error registering DID`, error)
+      return {
+        didDocumentMetadata: {},
+        didRegistrationMetadata: {},
+        didState: {
+          state: 'failed',
+          reason: `unknownError: ${error.message}`,
+        },
+      }
+    }
+  }
+
+  public async update(agentContext: AgentContext, options: WebvhDidUpdateOptions): Promise<DidUpdateResult> {
+    const didRepository = agentContext.dependencyManager.resolve(DidRepository)
+
+    try {
+      if (!options.didDocument) {
+        return {
+          didDocumentMetadata: {},
+          didRegistrationMetadata: {},
+          didState: {
+            state: 'failed',
+            reason: 'Provide a valid didDocument',
+          },
+        }
+      }
+
+      // Update the DID using didwebvh-ts
+      const response = await updateDID({doc: options.didDocument.toJSON()})
+      
+      // Update the did record
+      const didRecord = await didRepository.findCreatedDid(agentContext, options.did)
+      if (didRecord) {
+        didRecord.didDocument = options.didDocument
+        await didRepository.update(agentContext, didRecord)
+      }
+
+      return {
+        didDocumentMetadata: {},
+        didRegistrationMetadata: {},
+        didState: {
+          state: 'finished',
+          did: options.did,
+          didDocument: options.didDocument,
+        },
+      }
+    } catch (error) {
+      agentContext.config.logger.error(`Error updating DID`, error)
+      return {
+        didDocumentMetadata: {},
+        didRegistrationMetadata: {},
+        didState: {
+          state: 'failed',
+          reason: `unknownError: ${error.message}`,
+        },
+      }
+    }
+  }
+
+  public async deactivate(agentContext: AgentContext, options: WebvhDidDeactivateOptions): Promise<DidDeactivateResult> {
+    // const didRepository = agentContext.dependencyManager.resolve(DidRepository)
```

If this is not supported yet we can return an error for now and remove this code